### PR TITLE
Add PCG segmentation based on simultaneous ECG + bug correction

### PR DESCRIPTION
**Pull request by sofia3ms**
Added a new function in the pcg module that segments the PCG into the heart sound states based on the R-peak and end T-wave positions of a simultaneous ECG signal (adapted to Python from code developed by [David Springer](https://github.com/davidspringer/Springer-Segmentation-Code) ). An example file with the two simultaneous signals is included.

I used the `getTPositions` function from the ecg module and found an issue. The identification of the R-peak and the T-wave on the ECG templates was done using the array indexes:

https://github.com/scientisst/BioSPPy/blob/7a34e1f1c01fc6c27e0620c2cb9e0b60c34ebc13/biosppy/signals/ecg.py#L1599-L1602

but these values seem to change depending on the sampling frequency of the signal. I changed it to be based on the templates time axis reference `templates_ts` in seconds:

```python
# R peak on the template is always on time instant 0 seconds
template_r_position = np.argmin(np.abs(templates_ts - 0))  
# the T wave start is approximately 0.14 seconds after R-peak
template_T_position_min = np.argmin(np.abs(templates_ts - 0.14))
```

I corrected the same issue in `getQPositions`, `getSPositions` and `getPPositions`, and also added some exception handling so the functions return an output even when they can't find the specific wave in all of the templates.

### CODE REVIEW

File: biosppy/signals/ecg.py [hunk 1]
```diff
@@ -1534,58 +1574,83 @@ def getPPositions(ecg_proc=None, show=False):
...
+            )
+        
+        plt.legend()
+        plt.show()
+        
+        P_positions = np.array(P_positions)
+        P_start_positions = np.array(P_start_positions)
+        P_end_positions = np.array(P_end_positions)
+        
+    return utils.ReturnTuple((P_positions, P_start_positions, P_end_positions,), ("P_positions","P_start_positions","P_end_positions",))
```
Comment (rafaelscsilva) [inline biosppy/signals/ecg.py:1643]:
```suggestion


```

File: biosppy/signals/pcg.py [hunk 2]
```diff
@@ -279,4 +302,155 @@ def identify_heart_sounds(beats = None, sampling_rate = 1000.):
     
     classification += 1    
         
-    return utils.ReturnTuple((classification,), ('heart_sounds',))
\ No newline at end of file
+    return utils.ReturnTuple((classification,), ('heart_sounds',))
+
+def ecg_based_segmentation(pcg_signal=None, ecg_signal=None,sampling_rate=1000.0,show=False):
```
Comment (rafaelscsilva) [inline biosppy/signals/pcg.py:307]:
```suggestion

def ecg_based_segmentation(pcg_signal=None, ecg_signal=None, sampling_rate=1000.0, show=False):
```

**Review by rafaelscsilva — COMMENTED** [review]
Thanks @sofia3ms for your PR and for fixing some of the bugs you found in the code! I made simple suggestions mainly about formatting the code, and the use of some variables. Let me know if you agree with them.

I would also suggest to change the filename `pcg+ecg.txt` to `pcg_ecg.txt` (or similar) to avoid using the `+` character.

File: biosppy/signals/pcg.py [hunk 3]
```diff
@@ -144,9 +147,13 @@ def find_peaks(signal=None,sampling_rate=1000.):
     return utils.ReturnTuple((peaks,envelope), 
                              ('peaks','homomorphic_envelope'))
 
-def homomorphic_filter(signal=None, sampling_rate=1000.):
+def homomorphic_filter(signal=None, sampling_rate=1000., f_LPF=8, order=1):
```
[the remark this answers is not in the document]
Comment (sofia3ms) [reply biosppy/signals/pcg.py:150]: Yes, great catch!

**Pull request merged**